packages/fxa-auth-server/lib/routes/subscriptions/stripe-webhook.ts

```diff
@@ -828,6 +833,18 @@ export class StripeWebhookHandler extends StripeHandler {
           });
         }
       }
+      // Add any valid plans found in Stripe that are missing from the cache.
+      // This is possible e.g. if a plan was created before a product's metadata
+      // was valid and was never updated afterward.
+      for (const plan of latestStripePlansForProduct) {
+        const cachedPlan = updatedPlans.find((p) => p.id === plan.id);
+        if (!cachedPlan) {
+          updatedPlans.push({
+            ...plan,
```

Comment:
Do we need to run the `subscriptionProductMetadataValidator` for the plan pulled from Stripe, similar to how the cachedPlans are validated with the product on lines 803-806?

```
await subscriptionProductMetadataValidator.validateAsync({
  ...product.metadata,
  ...plan.metadata,
});
```

Reply:
They're already validated in `fetchAllPlans` above.

Reply:
Ah nice, I didn't realize `fetchAllPlans` validates the plan.

That said, I believe `subscriptionproductMetadataValidator` validates the Product metadata, with the Plan metadata, for each of the plans belonging to the Product.

At the moment validation is done for each of the `cachedPlans`, but not for any of the plans that are in Stripe, but missing from  `cachedPlans`.

Reply:
> At the moment validation is done for each of the `cachedPlans`, but not for any of the plans that are in Stripe, but missing from `cachedPlans`.

Validation is done for all of the plans in Stripe, including those not in the cache thanks to [`fetchAllPlans`](https://github.com/mozilla/fxa/blob/0ce0b6848055b776031ea20983c100a759907778/packages/fxa-shared/payments/stripe.ts#L322) (by way of [`validatePlan`](https://github.com/mozilla/fxa/blob/0ce0b6848055b776031ea20983c100a759907778/packages/fxa-auth-server/lib/payments/stripe.ts#L247) which it calls).

Reply:
Ah, I see now. Thank you for walking me through it quickly.

Reply:
Thank you for taking a careful look to make sure it made sense to you as well.